### ISSUE TITLE
Restructure Output Conditionals

### ISSUE THREAD

**Issue by mr-martian** [apertium/apertium-recursive#4]
Right now, output conditionals are compiled on the input side and so cannot access chunk tags set higher up the tree. Fixing this will probably require moving the chunk surface information from ```OutputChunk``` to ```Rule```.

@jonorthwash

**Comment by mr-martian** [comment]
This change will introduce a very specific caveat:

```
NP clitic -> n clitic adj { { 1 _1 3 } _2 2 } (condition)
                          { { 1 _2 3 } _1 2 } ;
```

When the conditionals are evaluated on the output side, the first chunk will receive the first or second blank as expected. However, since the position of the blank that's outside the chunk has to be resolved on the input side, it will always be the second blank because I can't think of any way to resolve this other than just going with the first pattern and putting warnings in the documentation.

**Comment by mr-martian** [comment]
Fixed by deabde144c4489d0085259714d9026015f6bb9f1

I'm not closing this yet due to the issue in the previous comment.

**Comment by mr-martian** [comment]
Due to a change in syntax, the above issue no longer exists.
